core/src/main/java/com/datastrato/gravitino/catalog/TopicOperationDispatcher.java

```diff
@@ -282,4 +269,95 @@ public boolean dropTopic(NameIdentifier ident) {
         ? droppedFromStore
         : droppedFromCatalog;
   }
+
+  private boolean importTopic(NameIdentifier identifier) {
+
+    EntityCombinedTopic combinedTopic = loadCombinedTopic(identifier);
+
+    if (combinedTopic.imported()) {
+      return false;
+    }
+
+    StringIdentifier stringId = getStringIdFromProperties(combinedTopic.topicProperties());
+
+    long uid;
+    if (stringId != null) {
+      // If the entity in the store doesn't match the external system, we use the data
+      // of external system to correct it.
+      uid = stringId.id();
+    } else {
+      // If store doesn't exist entity, we sync the entity from the external system.
+      uid = idGenerator.nextId();
+    }
+
+    TopicEntity topicEntity =
+        TopicEntity.builder()
+            .withId(uid)
+            .withName(combinedTopic.name())
+            .withComment(combinedTopic.comment())
+            .withNamespace(identifier.namespace())
+            .withAuditInfo(
+                AuditInfo.builder()
+                    .withCreator(combinedTopic.auditInfo().creator())
+                    .withCreateTime(combinedTopic.auditInfo().createTime())
+                    .withLastModifier(combinedTopic.auditInfo().lastModifier())
+                    .withLastModifiedTime(combinedTopic.auditInfo().lastModifiedTime())
+                    .build())
+            .build();
+
+    try {
+      store.put(topicEntity, true);
+    } catch (Exception e) {
+      LOG.error(FormattedErrorMessages.STORE_OP_FAILURE, "put", identifier, e);
+      throw new RuntimeException("Fail to access underlying storage");
+    }
+
+    return true;
+  }
+
+  private boolean isEntityExist(NameIdentifier ident) {
```

Comment:
Can we share this method between table, schema, topic and so on?

Reply:
OK.